src/main/java/org/colonelkai/publictransit/TravelManager.java

```diff
@@ -0,0 +1,132 @@
+package org.colonelkai.publictransit;
+
+import org.colonelkai.publictransit.line.travel.Travel;
+import org.core.TranslateCore;
+import org.core.entity.living.human.player.LivePlayer;
+import org.core.entity.living.human.player.User;
+import org.core.schedule.Scheduler;
+import org.core.schedule.unit.TimeUnit;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+public class TravelManager {
+    private final HashMap<Travel, Scheduler> travelMap = new HashMap<>();
+
+    public Optional<Travel> getTravelFromPlayer(UUID uuid) {
+        return this.travelMap
+                .keySet()
+                .parallelStream()
+                .filter(
+                        travel -> travel.getPlayer().equals(uuid)
+                )
+                .findAny();
+    }
+
+    public Optional<Travel> getTravelFromPlayer(User player) {
+        return this.getTravelFromPlayer(player.getUniqueId());
+    }
+
+    private Optional<Travel> getTravelFromScheduler(Scheduler scheduler) {
+        return this.travelMap.entrySet()
+                .parallelStream()
+                .filter((e) -> e.getValue().equals(scheduler))
+                .map(Map.Entry::getKey)
+                .findAny();
+    }
+
+    public Scheduler getScheduler(Travel travel) {
+        return this.travelMap.get(travel);
+    }
+
+    public void updateScheduler(Travel travel, Scheduler newScheduler) {
+        this.travelMap.replace(travel, newScheduler);
+    }
+
+    public void stopTravel(Travel travel, boolean returnToOriginal) {
+        this.travelMap.remove(travel);
+
+        if(returnToOriginal) {
+            TranslateCore.getScheduleManager().schedule()
+                    .setDelay(5)
+                    .setDelayUnit(TimeUnit.MINECRAFT_TICKS)
+                    .setRunner(scheduler -> {
+                        Optional<LivePlayer> optionalLivePlayer
+                                = TranslateCore.getServer().getOnlinePlayers()
+                                .parallelStream()
+                                .filter(p -> p.getUniqueId().equals(travel.getPlayer()))
+                                .findAny();
+                        if(optionalLivePlayer.isEmpty()) {
+                            return;
+                        }
+                        LivePlayer player = optionalLivePlayer.get();
+
+                        player.setPosition(travel.getStartingNode().getPosition());
```

Comment:
From my understand of this, if the player disconnects, you then schedule every 5 minecraft ticks to check if they have rejoined and if so then teleport them back to the travel start?

Wouldnt it be more logical to set the position on player join back to the starting position? or better yet, set the offline player location (I know its not in the translation layer .... ill check that)

Reply:
> or better yet, set the offline player location (I know its not in the translation layer .... ill check that) (paper doesnt support setting position of offline players .... why????)
I didn't know that was an option, and I guess it's not 😆 

> Wouldnt it be more logical to set the position on player join back to the starting position?
ooh yeah, I'll do this instead in a moment.

Reply:
The players location will be stored somewhere (likely player data in the world folder) so you should be able to modify it. Sponge actually allows the manipulation of the position and world (as a resource key). Just paper doesn't